carbonmark-api/src/server.ts

```diff
@@ -0,0 +1,18 @@
+import fastify from "fastify";
+import app from "./app";
+
+const PORT = 3003;
+
+async function main() {
```

Comment:
Can you add a comment explaining that when/why this file exists? Looks like it's only for the `dev` command, unclear if thats used for running tests as well

Reply:
Sure thing, added some docs. 

```ts
/**
 * This server is required for local development because the API is deployed
 * via Vercel serverless which handles incoming requests. By separating concerns (server & app)
 * we can easily switch deployment environments and test.
 *
 * **Note** This is not required for tests as we use `app.inject` to mock requests to the FastifyInstance
 */
```